mqtt-client-Electron/renderer.js

```diff
@@ -6,52 +6,115 @@
 // process.
 let client = null
 
+/**
+ * this demo uses EMQX Public MQTT Broker (https://www.emqx.com/en/mqtt/public-mqtt5-broker), here are the details:
+ *
+ * Broker host: broker.emqx.io
+ * TCP Port: 1883
+ * SSL/TLS Port: 8883
+ * WebSocket port: 8083
+ * WebSocket over TLS/SSL port: 8084
+ */
+
 const options = {
   keepalive: 30,
   protocolId: 'MQTT',
   protocolVersion: 4,
   clean: true,
-  reconnectPeriod: 1000,
-  connectTimeout: 30 * 1000,
-  will: {
-    topic: 'WillMsg',
-    payload: 'Connection Closed abnormally..!',
-    qos: 0,
-    retain: false
-  },
-  rejectUnauthorized: false
+  connectTimeout: 30 * 1000, // ms
+  reconnectPeriod: 4000, // ms
+  // for more options, please refer to https://github.com/mqttjs/MQTT.js#mqttclientstreambuilder-options
 }
+
+const protocolSelect = document.getElementById('protocolSelect')
 const connectBtn = document.getElementById('connectBtn')
 const disconnectBtn = document.getElementById('disconnectBtn')
 const subBtn = document.getElementById('subBtn')
 const unsubBtn = document.getElementById('unsubBtn')
 const sendBtn = document.getElementById('sendBtn')
 
+protocolSelect.addEventListener('change', onProtocolChange)
 connectBtn.addEventListener('click', onConnect)
 disconnectBtn.addEventListener('click', onDisconnect)
 subBtn.addEventListener('click', onSub)
 unsubBtn.addEventListener('click', onUnsub)
 sendBtn.addEventListener('click', onSend)
 
+//  protocol->port    mqtt: 1883; mqtts: 8883; ws: 8083; wss: 8084
+function onProtocolChange() {
+  switch (connection.protocol.value) {
+    case 'mqtt':
+      connection.port.value = 1883
+      break;
+    case 'mqtts':
+      connection.port.value = 8883
+      break;
+    case 'ws':
+      connection.port.value = 8083
+      break;
+    case 'wss':
+      connection.port.value = 8084
+      break;
+    default:
+      break;
+  }
+}
```

Comment:
I don't recommend handling port numbers this way, because these are default values that users can modify, especially for cloud deployments, right? @0721Betty

Reply:
这里等于用户在切换协议的时候，自动适配我们公共broker的端口。但是用户仍然可以在页面中修改端口的input框。

就相当于我们host字段默认是broker.emqx.io，但是用户也可以在input框中输入自己的host，这样。

Reply:
OK. I understand. And please use English in Public Repo. How about the previous comment about handling the path field? It's need to change?

Reply:
hide path form item, if protocol is not ws or wss?